### ISSUE TITLE
Wrong (weird) Estimated query execution time (max_execution_time / max_execution_speed)

### ISSUE THREAD

**Issue by den-crane** [ClickHouse/ClickHouse#6417]
19.13.1.1010

create table ES(A String) Engine=MergeTree order by tuple();
insert into ES select toString(number)  from numbers(10000000);
SET max_execution_time = 1, max_execution_speed = 1000000;
SET max_threads=1
set max_block_size=1000000;

SELECT * FROM ES LIMIT 1 format Null;
DB::Exception: Estimated query execution time (1003499.9296 seconds) is too long. 
Maximum: 1. Estimated rows to process: 10035200.

LIMIT 10
DB::Exception: Estimated query execution time (8027.156479999999 seconds) is too long. 
Maximum: 1. Estimated rows to process: 10035200.

LIMIT 100
DB::Exception: Estimated query execution time (1605.531648 seconds) is too long. 
Maximum: 1. Estimated rows to process: 10035200.

LIMIT 1000
DB::Exception: Estimated query execution time (602.1019648 seconds) is too long. 
Maximum: 1. Estimated rows to process: 10035200.

LIMIT 10000
B::Exception: Estimated query execution time (4.01307648 seconds) is too long. 
Maximum: 1. Estimated rows to process: 10035200.

LIMIT 100000
B::Exception: Estimated query execution time (4.013979648 seconds) is too long. 
Maximum: 1. Estimated rows to process: 10035200.

LIMIT 1000000
1000000 rows in set. Elapsed: 1.125 sec. 
Processed 1.12 million rows, 16.85 MB (997.34 thousand rows/s., 14.97 MB/s.)

------------
set max_block_size=1024

LIMIT 1
1 rows in set. Elapsed: 0.010 sec.

LIMIT 10000
DB::Exception: Estimated query execution time (13.020744 seconds) is too long. 
Maximum: 1. Estimated rows to process: 10002432.

LIMIT 100000
100000 rows in set. Elapsed: 0.005 sec. 
Processed 111.62 thousand rows, 1.73 MB (23.74 million rows/s., 368.77 MB/s.)

**Comment by den-crane** [comment]
still relevant

SELECT *
FROM ES
LIMIT 10
FORMAT Null

↘ Progress: 0.00 rows, 0.00 B (0.00 rows/s., 0.00 B/s.) Received exception from server (version 19.16.1):
Code: 160. DB::Exception: Received from localhost:9000. DB::Exception: Estimated query execution time (3999.9725568 seconds) is too long. Maximum: 1. Estimated rows to process: 10002432.

**Comment by alexey-milovidov** [comment]
The issue is no longer relevant, because in recent versions (since 19.17) the estimation (prediction) of query execution time is performed only after `timeout_before_checking_execution_speed` has been passed and only if this setting is set.

But I can still reproduce the same effect if I will set it to very low value:

`SET timeout_before_checking_execution_speed = 0.0001`

```
SELECT *
FROM ES
LIMIT 1
FORMAT Null

↗ Progress: 1.00 rows, 10.00 B (9.88 rows/s., 98.83 B/s.)  0%Received exception from server (version 20.5.1):
Code: 160. DB::Exception: Received from localhost:9000. DB::Exception: Estimated query execution time (25159.532544 seconds) is too long. Maximum: 1. Estimated rows to process: 6291456: While executing MergeTree.
```

I will add your test case.

**Comment by alexey-milovidov** [comment]
```
0%Received
```

I will also fix bad formatting of output in clickhouse-client.

**Comment by alexey-milovidov** [comment]
> But I can still reproduce the same effect if I will set it to very low value

And this behaviour is 100% legitimate.

**Comment by alexey-milovidov** [comment]
I have added a test.
